src/Microsoft.AspNetCore.DataProtection/TypeForwardingActivator.cs

```diff
@@ -39,16 +40,14 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
                 forwardedTypeName = originalTypeName.Replace(OldNamespace, CurrentNamespace);
             }
 
-#if NET46
-            if (candidate || forwardedTypeName.Contains(CurrentNamespace))
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Core"))
```

Comment:
This looks like a no-go to me. At the very least, the check above isn't using the proper culture, but this still seems like a wrong way to check what we want here. Is there an actual reliable way to check the platform that doesn't use strings?